media/js/mozorg/plugincheck-update.js

```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function($) {
+    'use strict';
+
+    var client = window.Mozilla.Client;
+    var body = $('body');
+
+    if (client._isFirefoxDesktop) {
+        client.getFirefoxDetails(function(details) {
+            if (details.isUpToDate) {
```

Comment:
Actually looking at the logic for `Mozilla.Client`, it detects ESR sucessfully too (using UITour) and compares that against the latest ESR version number in product details, so you can actually omit the ESR check and it should "just work".

We probably don't want to show this to out-of-date Beta/Nightly users though, so maybe just check that the channel is either 'release' or 'esr'?